### PR TITLE
zgrab2: init at 20210327-17a5257

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Web application scanner

https://github.com/zmap/zgrab2

Related to #81418

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26692,6 +26692,8 @@ in
     autoconf = buildPackages.autoconf269;
   };
 
+  zgrab2 = callPackage ../tools/security/zgrab2 {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26695]:
```suggestion
  zgrab2 = callPackage ../tools/security/zgrab2 { };
```

File: pkgs/tools/security/zgrab2/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "zmap";
+    repo = pname;
+    inherit rev;
+    sha256 = "1hxk2jggj8lww97lwmks46i001p5ycnxnck8yya6d0fd3ayxvw2w";
+  };
+
+  vendorSha256 = "1s0azy5b5hi5h24vs6a9f1n70l980vkid28ihqh10zq6ajmds2z3";
+
+  subPackages = "cmd/zgrab2";
```
Comment (SuperSandro2000) [inline pkgs/tools/security/zgrab2/default.nix:20]:
```suggestion
  subPackages = [ "cmd/zgrab2" ];
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/top-level/all-packages.nix]: Update, thanks.

File: pkgs/tools/security/zgrab2/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/security/zgrab2/default.nix]: Updated, thanks.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 117739` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zgrab2</li>
  </ul>
</details>

**Pull request merged**